docs/source/guide/distributed-partition.rst

```diff
@@ -0,0 +1,307 @@
+not only saves nodes/edges it owns, but also includes node/edges that are adjacent to
+the partition (called **HALO** nodes/edges). The ``inner_node`` and ``inner_edge``
+indicate whether a node/edge truely belongs to the partition (value is ``True``)
+or is a HALO node/edge (value is ``False``).
+
+The :func:`~dgl.distributed.load_partition` function loads all data at once. Users can
+load features or the partition book using the :func:`dgl.distributed.load_partition_feats`
+and :func:`dgl.distributed.load_partition_book` APIs respectively.
+
+
+Parallel METIS partitioning
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For massive graphs where parallel preprocessing is desired, DGL supports
+`ParMETIS <http://glaros.dtc.umn.edu/gkhome/metis/parmetis/overview>`__ as one
+of the choices of partitioning algorithms.
+
+.. note::
+
+    Because ParMETIS does not support heterogeneous graph, users need to
+    conduct ID conversion before and after running ParMETIS.
+    Check out chapter :ref:`guide-distributed-hetero` for explanation.
+
+.. note::
+
+    Please make sure that the input graph to ParMETIS does not have
+    duplicate edges (or parallel edges) or self-loop edges.
+
+ParMETIS Installation
+^^^^^^^^^^^^^^^^^^^^^^
+ParMETIS requires METIS and GKLib. Please follow the instructions `here
+<https://github.com/KarypisLab/GKlib>`__ to compile and install GKLib. For
+compiling and install METIS, please follow the instructions below to clone
+METIS with GIT and compile it with int64 support.
+
+.. code-block:: bash
+
+    git clone https://github.com/KarypisLab/METIS.git
+    make config shared=1 cc=gcc prefix=~/local i64=1
+    make install
+
+
+For now, we need to compile and install ParMETIS manually. We clone the DGL branch of ParMETIS as follows:
+
+.. code-block:: bash
+
+    git clone --branch dgl https://github.com/KarypisLab/ParMETIS.git
+
+Then compile and install ParMETIS.
+
+.. code-block:: bash
+
+    make config cc=mpicc prefix=~/local
+    make install
+
+Before running ParMETIS, we need to set two environment variables: ``PATH`` and ``LD_LIBRARY_PATH``.
+
+.. code-block:: bash
+
+    export PATH=$PATH:$HOME/local/bin
+    export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$HOME/local/lib/
+
+Input format
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The input graph for ParMETIS is stored in three files with the following names:
+``xxx_nodes.txt``, ``xxx_edges.txt`` and ``xxx_stats.txt``, where ``xxx`` is a
+graph name.
+
+Each row in ``xxx_nodes.txt`` stores the information of a node with the following format:
+
+.. code-block:: none
+
+    <node_type> <weight1> ... <orig_type_node_id>
+
+All fields are separated by whitespace:
+
+* ``<node_type>`` is an integer. For a homogeneous graph, its value is always
+  0. For heterogeneous graphs, its value indicates the type of each node.
+* ``<weight1>``, ``<weight2>``, etc. are integers that indicate the node weights
+  used by ParMETIS to balance graph partitions. If a user does not provide node
```

Comment:
The weights here is a little bit confusing. It is more like node groups that ParMETIS will try to balance the number of nodes within a group across different partitions.